### ISSUE TITLE
QUESTION: how to search order (in Woo admin) by custom order number or customer name?

### ISSUE THREAD

**Issue by tictok** [10up/ElasticPress#2151]
**Describe your question**
How can I get Elasticpress to search orders (in WooCommerce admin) by order meta?

I'm trying to use Elasticpress to speed up woocommerce admin, particularly searching and sorting orders (approx 60k orders).

- Searching by order ID works great (quickly).

- Searching for an order by Sequential order number (Woocommerce Sequential Order Numbers Pro) returns zero result. The sequential order number is stored in wp_postmeta table, meta key _order_number_formatted

I've tried whitelisting '_order_number_formatted' meta key which stores the Sequential Order Numbers Pro order number in wp_postmeta table via this method: [2014](https://github.com/10up/ElasticPress/issues/2014#issuecomment-753549461)

Anything else I can try to allow searching orders by custom / Sequential Order Numbers Pro or customer name?

Many thanks

**Comment by felipeelia** [comment]
Hi @tictok ,

Can you please try the following code:

```
function ep_allow_seq_order_number_key( $allowed ) {
	$allowed[] = '_order_number_formatted';
	return $allowed;

}
add_filter( 'ep_prepare_meta_allowed_protected_keys', 'ep_allow_seq_order_number_key' );

function ep_add_custom_field_ep_weighting( $fields, $post_type ) {
	if ( 'product' === $post_type ) {
		if ( empty( $fields['meta'] ) ) {
			$fields['meta'] = array(
				'label'    => 'Custom Fields',
				'children' => array(),
			);
		}

		$key = 'meta._order_number_formatted.value';

		$fields['meta']['children'][ $key ] = array(
			'key'   => $key,
			'label' => __( 'Sequential Order Number', 'textdomain' ),
		);
	}

	return $fields;
}
add_filter(
	'ep_weighting_fields_for_post_type',
	'ep_add_custom_field_ep_weighting',
	10,
	2
);
```

It will enable the new field in the Weighting Dashboard. So, you'll have to put this code in your theme's functions.php (or in a separate plugin), go to the Weighting Dashboard (inside the WP Dashboard -> ElasticPress -> Weighting), and enable the new field in the "Products" box. Don't forget to click on Save changes. After that, you'll have to reindex your content and it should work.

Can you please try that and tell us if it worked? Thanks!

**Comment by tictok** [comment]
brilliant - thank you. I'll try it a little later this evening when our store is quiet and report back.

**Comment by felipeelia** [comment]
As we didn’t have any movement in this issue for a while now, we are now closing it. If you’re still facing the issue, please let us know. Otherwise, if you’ve managed to get an answer to your problem, we’d highly appreciate it if you could get back to us here and leave the solution in a comment. Thanks!

**Comment by tictok** [comment]
Hi - apologies for not getting back to you Felipeelia. I started testing the snippet you provided, the option in the weighting dashboard was available but then ran into an unrelated memory issue with ElasticSearch when indexing that my host should be sorting out soon.   Thanks.

**Comment by millionleaves** [comment]
I can confirm that this snippet worked on a couple of sites I manage, but only if I also included the snippet from https://github.com/10up/ElasticPress/issues/2014#issuecomment-753549461. I use the WooCommerce Sequential Order Numbers plugin from SkyVerge, so my code looks like this:

`/* add custom order id in Elastic search index list */
add_filter( 'ep_prepare_meta_whitelist_key', function( $allow, $meta_key ){
	$meta_keys = ['_order_number']; // meta keys you want to allow
	return in_array( $meta_key, $meta_keys );
}, 10, 2 );
add_filter( 'ep_search_fields', 'update_search_fields', 11, 2 );

function update_search_fields( array $fields, array $args ): array {
$fields[] = 'meta._order_number.value';

return $fields;
}


function ep_allow_seq_order_number_key( $allowed ) {
	$allowed[] = '_order_number';
	return $allowed;

}
add_filter( 'ep_prepare_meta_allowed_protected_keys', 'ep_allow_seq_order_number_key' );

function ep_add_custom_field_ep_weighting( $fields, $post_type ) {
	if ( 'product' === $post_type ) {
		if ( empty( $fields['meta'] ) ) {
			$fields['meta'] = array(
				'label'    => 'Custom Fields',
				'children' => array(),
			);
		}

		$key = 'meta._order_number_formatted.value';

		$fields['meta']['children'][ $key ] = array(
			'key'   => $key,
			'label' => __( 'Sequential Order Number', 'textdomain' ),
		);
	}

	return $fields;
}
add_filter(
	'ep_weighting_fields_for_post_type',
	'ep_add_custom_field_ep_weighting',
	10,
	2
);`

**Comment by razorfrog** [comment]
I'm using "WooCommerce Sequential Order Numbers Pro" and am seeing this issue as well. I can search by actual order post ID, but not the sequential one that it's given by the plugin. 

I've added the code from @felipeelia and @millionleaves above and now see the Sequential Order Number field under search weighting, and I've reindexed. But there's no change in search results.

How can I troubleshoot this further?
